### PR TITLE
Change host metrics receiver to use the new metrics internal structs

### PR DESCRIPTION
**Pull request by bogdandrutu**
The most important changes are in the metadata files, others are just simple renaming.

Updates #1657

Depends on https://github.com/open-telemetry/opentelemetry-collector/pull/1684 which adds the SetDataType and adds tests for it.

### CODE REVIEW

File: receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_metadata.go [hunk 1]
```diff
@@ -35,22 +35,31 @@ const (
...
-	descriptor.SetUnit("bytes")
-	descriptor.SetType(dataold.MetricTypeInt64)
-	return descriptor
+var fileSystemUsageDescriptor = func() pdata.Metric {
+	metric := pdata.NewMetric()
+	metric.InitEmpty()
+	metric.SetName("system.filesystem.usage")
+	metric.SetDescription("Filesystem bytes used.")
+	metric.SetUnit("bytes")
+	metric.SetDataType(pdata.MetricDataTypeIntSum)
```
Comment (james-bebbington) [inline receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_metadata.go:44]: Should be Gauge

File: receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_metadata.go [hunk 2]
```diff
@@ -35,22 +35,31 @@ const (
...
-	descriptor.SetUnit("1")
-	descriptor.SetType(dataold.MetricTypeInt64)
-	return descriptor
+var fileSystemINodesUsageDescriptor = func() pdata.Metric {
+	metric := pdata.NewMetric()
+	metric.InitEmpty()
+	metric.SetName("system.filesystem.inodes.usage")
+	metric.SetDescription("FileSystem operations count.")
+	metric.SetUnit("1")
+	metric.SetDataType(pdata.MetricDataTypeIntSum)
```
Comment (james-bebbington) [inline receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_metadata.go:58]: Should be Gauge

File: receiver/hostmetricsreceiver/internal/scraper/memoryscraper/memory_metadata.go [hunk 3]
```diff
@@ -36,12 +36,16 @@ const (
...
-	descriptor.SetUnit("bytes")
-	descriptor.SetType(dataold.MetricTypeInt64)
-	return descriptor
+var memoryUsageDescriptor = func() pdata.Metric {
+	metric := pdata.NewMetric()
+	metric.InitEmpty()
+	metric.SetName("system.memory.usage")
+	metric.SetDescription("Bytes of memory in use.")
+	metric.SetUnit("bytes")
+	metric.SetDataType(pdata.MetricDataTypeIntSum)
```
Comment (james-bebbington) [inline receiver/hostmetricsreceiver/internal/scraper/memoryscraper/memory_metadata.go:45]: Should be Gauge

File: receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_metadata.go [hunk 4]
```diff
@@ -35,52 +35,72 @@ const (
...
-	descriptor.SetUnit("bytes")
-	descriptor.SetType(dataold.MetricTypeInt64)
-	return descriptor
+var networkTCPConnectionsDescriptor = func() pdata.Metric {
+	metric := pdata.NewMetric()
+	metric.InitEmpty()
+	metric.SetName("system.network.tcp_connections")
+	metric.SetDescription("The number of tcp connections")
+	metric.SetUnit("bytes")
+	metric.SetDataType(pdata.MetricDataTypeIntSum)
```
Comment (james-bebbington) [inline receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_metadata.go:100]: Should be Gauge

File: receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_metadata.go [hunk 5]
```diff
@@ -15,27 +15,35 @@
...
-	descriptor.SetUnit("1")
-	descriptor.SetType(dataold.MetricTypeInt64)
-	return descriptor
+var processesRunningDescriptor = func() pdata.Metric {
+	metric := pdata.NewMetric()
+	metric.InitEmpty()
+	metric.SetName("system.processes.running")
+	metric.SetDescription("Total number of running processes.")
+	metric.SetUnit("1")
+	metric.SetDataType(pdata.MetricDataTypeIntSum)
```
Comment (james-bebbington) [inline receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_metadata.go:29]: Should be Gauge

File: receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_metadata.go [hunk 6]
```diff
@@ -15,27 +15,35 @@
...
-	descriptor.SetUnit("1")
-	descriptor.SetType(dataold.MetricTypeInt64)
-	return descriptor
+var processesBlockedDescriptor = func() pdata.Metric {
+	metric := pdata.NewMetric()
+	metric.InitEmpty()
+	metric.SetName("system.processes.blocked")
+	metric.SetDescription("Total number of blocked processes.")
+	metric.SetUnit("1")
+	metric.SetDataType(pdata.MetricDataTypeIntSum)
```
Comment (james-bebbington) [inline receiver/hostmetricsreceiver/internal/scraper/processesscraper/processes_metadata.go:43]: Should be Gauge

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/process_metadata.go [hunk 7]
```diff
@@ -42,42 +42,58 @@ const (
...
-	descriptor.SetUnit("bytes")
-	descriptor.SetType(dataold.MetricTypeInt64)
-	return descriptor
+var physicalMemoryUsageDescriptor = func() pdata.Metric {
+	metric := pdata.NewMetric()
+	metric.InitEmpty()
+	metric.SetName("process.memory.physical_usage")
+	metric.SetDescription("The amount of physical memory in use.")
+	metric.SetUnit("bytes")
+	metric.SetDataType(pdata.MetricDataTypeIntSum)
```
Comment (james-bebbington) [inline receiver/hostmetricsreceiver/internal/scraper/processscraper/process_metadata.go:65]: Should be Gauge

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/process_metadata.go [hunk 8]
```diff
@@ -42,42 +42,58 @@ const (
...
-	descriptor.SetUnit("bytes")
-	descriptor.SetType(dataold.MetricTypeInt64)
-	return descriptor
+var virtualMemoryUsageDescriptor = func() pdata.Metric {
+	metric := pdata.NewMetric()
+	metric.InitEmpty()
+	metric.SetName("process.memory.virtual_usage")
+	metric.SetDescription("Virtual memory size.")
+	metric.SetUnit("bytes")
+	metric.SetDataType(pdata.MetricDataTypeIntSum)
```
Comment (james-bebbington) [inline receiver/hostmetricsreceiver/internal/scraper/processscraper/process_metadata.go:79]: Should be Gauge

File: receiver/hostmetricsreceiver/internal/scraper/swapscraper/swap_metadata.go [hunk 9]
```diff
@@ -49,32 +49,44 @@ const (
...
-	descriptor.SetUnit("pages")
-	descriptor.SetType(dataold.MetricTypeInt64)
-	return descriptor
+var swapUsageDescriptor = func() pdata.Metric {
+	metric := pdata.NewMetric()
+	metric.InitEmpty()
+	metric.SetName("system.swap.usage")
+	metric.SetDescription("Swap (unix) or pagefile (windows) usage.")
+	metric.SetUnit("pages")
+	metric.SetDataType(pdata.MetricDataTypeIntSum)
```
Comment (james-bebbington) [inline receiver/hostmetricsreceiver/internal/scraper/swapscraper/swap_metadata.go:58]: Should be Gauge

File: receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_metadata.go [hunk 10]
```diff
@@ -35,22 +35,31 @@ const (
...
+	sum := metric.IntSum()
+	sum.InitEmpty()
+	// TODO: Investigate why not cumulative?
```
Comment (james-bebbington) [inline receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_metadata.go:61]: I believe inodes can be freed?

**Review by james-bebbington — APPROVED** [review]
Nvm - ignore all my previous comments.

The distinction between a Gauge and a non-monotomic Cumulative is:

- A Cumulative can be used in any case where measurements are added: e.g. when data is written to disk, +some bytes, when data is deleted from disk, -some bytes
- A Gauge is reserved for cases where the data source is not a sum or is not known, e.g. CPU Load (which is a moving average)

With that context, this all looks good.

File: receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_metadata.go (hunk 10 above)
Comment (bogdandrutu) [reply receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_metadata.go:61]: Are you asking... Don't know that's what I left the Todo, based on the description this is the number of ops so it is cumulative. Maybe you can check :)

Comment (james-bebbington) [reply receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_metadata.go:61]: Yes that's what I was answering.

And oh yea the description is wrong 😮  Seems like that was copied from the disk operations metric by mistake. It should be "iNodes used."

Comment (bogdandrutu) [reply receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_metadata.go:61]: Done.

**Pull request merged**